File: php/proxy-http.php
<?php
$username = 'USERNAME';
$password = 'PASSWORD';
$proxy = 'gw-us.nstproxy.com:24125';
$query = curl_init('API_URL');
curl_setopt($query, CURLOPT_RETURNTRANSFER, 1);
curl_setopt($query, CURLOPT_PROXY, "http://$proxy");
curl_setopt($query, CURLOPT_PROXYUSERPWD, "$username:$password");
$output = curl_exec($query);
curl_close($query);
if ($output)
    echo $output;
?>

File: java/proxy-http.java
package example;

import org.apache.http.HttpHost;
import org.apache.http.client.fluent.*;

public class Example {
    public static void main(String[] args) throws Exception {
        HttpHost entry = new HttpHost("gw-us.nstproxy.com", 24125);
        String query = Executor.newInstance()
            .auth(entry, "username", "password")
            .execute(Request.Get("API_URL").viaProxy(entry))
            .returnContent().asString();
        System.out.println(query);
    }
}